### PR TITLE
Allow org admins to toggle source data between tnds and bods

### PR DESCRIPTION
**Pull request by alvak1**
Testing instructions

- Login as org admin user and then access the Account Settings page from top right corner. The radio buttons should be displayed only to org admins
- After the source data is clicked - navigate to consequence services page and observe that the services data changes based on the source for mode of transport
- The services dropdown in view  all disruptions page should display both tnds and bods data

### CODE REVIEW

File: site/components/map/ServicesMap.tsx [hunk 1]
```diff
@@ -41,7 +45,7 @@ interface MapProps {
     stateUpdater: Dispatch<SetStateAction<PageState<Partial<ServicesConsequence>>>>;
     state: PageState<Partial<ServicesConsequence>>;
     searchedRoutes?: Partial<(Routes & { serviceId: number })[]>;
-    services?: Service[];
+    services?: Service[] | null;
```
Comment (alainajaffer) [inline site/components/map/ServicesMap.tsx:48]: why was this added?

File: site/interfaces/index.ts [hunk 2]
```diff
@@ -68,7 +68,8 @@ export interface PageState<T> {
 }
 
 export interface CreateConsequenceProps {
-    initialServices?: Service[];
+    initialBodsServices?: Service[] | null;
```
Comment (alainajaffer) [inline site/interfaces/index.ts:71]: why is this | null can it not just be optional and  Service[]?

File: site/utils/formUtils.ts [hunk 3]
```diff
@@ -84,3 +85,29 @@ export const getStopType = (stopType: string | undefined) => {
         return "Stop";
     }
 };
+
+// eslint-disable-next-line @typescript-eslint/require-await
+export const filterServices = async (servicesData: Service[] | undefined): Promise<Service[] | null> => {
+    return new Promise((resolve) => {
```
Comment (alainajaffer) [inline site/utils/formUtils.ts:91]: why return a promise here?

File: site/utils/formUtils.ts [hunk 4]
```diff
@@ -84,3 +85,29 @@ export const getStopType = (stopType: string | undefined) => {
...
+        let services: Service[] | null = null;
```
Comment (alainajaffer) [inline site/utils/formUtils.ts:92]: why not make this an empty array?

File: site/utils/formUtils.ts [hunk 5]
```diff
@@ -84,3 +85,29 @@ export const getStopType = (stopType: string | undefined) => {
         return "Stop";
     }
 };
+
+// eslint-disable-next-line @typescript-eslint/require-await
+export const filterServices = async (servicesData: Service[] | undefined): Promise<Service[] | null> => {
```
Comment (alainajaffer) [inline site/utils/formUtils.ts:90]: why not make this an optional parameter instead of | undefined

File: site/components/map/ServicesMap.tsx (hunk 1 above)
Comment (alvak1) [reply site/components/map/ServicesMap.tsx]: done - removed null

File: site/utils/formUtils.ts (hunk 3 above)
Comment (alvak1) [reply site/utils/formUtils.ts]: done - removed promise as its implicitly added

File: site/utils/formUtils.ts (hunk 4 above)
Comment (alvak1) [reply site/utils/formUtils.ts]: done - updated to empty array

File: shared-ts/enums.ts [hunk 6]
```diff
@@ -176,3 +176,8 @@ export enum SocialMediaPostStatus {
     successful = "Successful",
     rejected = "Rejected",
 }
+
```
Comment (alainajaffer) [inline shared-ts/enums.ts:179]: sometimes when i change a mode of transport the services / stops clear - is this expected?

File: site/utils/formUtils.ts [hunk 7]
```diff
@@ -84,3 +85,27 @@ export const getStopType = (stopType: string | undefined) => {
         return "Stop";
     }
 };
+
+// eslint-disable-next-line @typescript-eslint/require-await
+export const filterServices = async (servicesData?: Service[]) => {
```
Comment (alainajaffer) [inline site/utils/formUtils.ts:90]: can you remove the async

Comment (alvak1) [reply site/utils/formUtils.ts:90]: If async was removed then the method would run synchronously and not in parallel

Comment (alainajaffer) [reply site/utils/formUtils.ts:90]: I don't think it will make too much of a difference to performance in this case

**Review by alainajaffer — APPROVED** [review]
